### PR TITLE
initial PWA lighthouse extension

### PR DESCRIPTION
**Pull request by samccone**
primarily a port of https://gist.github.com/PaulKinlan/ce7c707e4cb96c752816 to work as an extension

ref #37

:rotating_light:

### CODE REVIEW

**Comment by samccone** [comment]
cc @PaulKinlan

File: extension/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "lighthouse",
+  "private": true,
+  "engines": {
+    "node": ">=0.8.0"
+  },
```
Comment (paulirish) [inline extension/package.json]: npm run build not yet setup to run gulp build

Comment (samccone) [inline extension/package.json]: whoops 1 sec

Comment (samccone) [inline extension/package.json]: resolved

File: extension/.babelrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (brendankenny) [inline extension/.babelrc]: hmm, why run it through the es2015 preset? Except for maybe the import this all looks like it'll run in stable chrome

Comment (samccone) [inline extension/.babelrc]: no harm transpiling really (plus this was the defaultttt) so :+1:

**Comment by paulirish** [comment]
In action: 
image
image
image

File: extension/.babelrc (hunk 2 above)
Comment (brendankenny) [inline extension/.babelrc]: well there's the compile step itself, and everything with babel in the name in here, and the npm dependencies... :P

but if it comes from the generator that may just be annoying to remove

File: package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "node": ">=4.0.0"
   },
   "scripts": {
-    "lint": "eslint .",
+    "lint": "eslint --ignore-path extension .",
```
Comment (brendankenny) [inline package.json]: you can just add `extension/` to `.eslintignore`

Comment (samccone) [inline package.json]: dope k

File: extension/app/scripts.babel/manifest-parser.js [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+  function _parseURL(args) {
+    var object = args.object;
+    var property = args.property;
+
+    var str = _parseString({object: object, property: property, trim: false});
+    if (str === undefined) {
+      return undefined;
+    }
+
+    return object[property];
```
Comment (brendankenny) [inline extension/app/scripts.babel/manifest-parser.js]: some of this stuff from the original bookmarklet is just crazy-town banana pants. Where does it come from originally, I wonder?

Comment (samccone) [inline extension/app/scripts.babel/manifest-parser.js]: i chose to do as little as possible to get this first iteration shipped :)

**Comment by samccone** [comment]
image

looking sharp

**Comment by paulirish** [comment]
@samccone can you note the items remaining that we'll resolve later in #38?

**Comment by samccone** [comment]
:+1: noted and lgtm

File: extension/.gitignore [hunk 5]
```diff
@@ -0,0 +1,10 @@
+node_modules
+temp
+.tmp
+dist
+.sass-cache
+app/bower_components
+test/bower_components
+package
+app/scripts
+
```
Comment (thediablohijodelucifer-ai) [inline extension/.gitignore]: 9

**Pull request merged**